src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -83,41 +126,51 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
+        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        {
+            if(string.IsNullOrWhiteSpace(framework))
```

Comment:
Here is an example of where you are assuming empty framework means the shared dependencies section. Where what should happen here is that we should always add this set of dependencies to every framework.